app/src/main/java/org/cru/godtools/ui/dashboard/tools/ToolsCategoryDataModel.kt

```diff
@@ -0,0 +1,43 @@
+package org.cru.godtools.ui.dashboard.tools
+
+import android.app.Application
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.map
+import dagger.hilt.android.lifecycle.HiltViewModel
+import java.util.Locale
+import javax.inject.Inject
+import org.ccci.gto.android.common.androidx.lifecycle.combineWith
+import org.ccci.gto.android.common.db.Expression.Companion.constants
+import org.ccci.gto.android.common.db.Query
+import org.ccci.gto.android.common.db.getAsLiveData
+import org.cru.godtools.base.ui.util.getCategory
+import org.cru.godtools.model.Tool
+import org.keynote.godtools.android.db.Contract
+import org.keynote.godtools.android.db.GodToolsDao
+
+@HiltViewModel
+class ToolsCategoryDataModel @Inject constructor(dao: GodToolsDao, context: Application) : AndroidViewModel(context) {
+
+    val selectedCategory = MutableLiveData<String?>(null)
+
+    private val allTools = Query.select<Tool>().where(
+        Contract.ToolTable.FIELD_TYPE.`in`(*constants(Tool.Type.TRACT, Tool.Type.ARTICLE, Tool.Type.CYOA))
+            .and(Contract.ToolTable.FIELD_HIDDEN.ne(true))
+    ).getAsLiveData(dao)
+
+    val categories = allTools.map {
+        it.mapNotNull { tool ->
+            tool.category?.let { category -> Pair(category, tool.getCategory(getApplication(), Locale.getDefault())) }
```

Comment:
what are you trying to do here?

Reply:
I am getting the category from the tool and the category that will be used in the view in one pair.

Reply:
It returns a `LiveData<List<Pair<String, String>>>`

Reply:
you shouldn't need the actual `tool` object to resolve the display name of the category

Reply:
`categories` should be a `LiveData<List<String>>`. the display name of the category should be handled within data binding using a `getToolCategory` method to convert the internal category name to a localized category name

Reply:
you will probably need to create a new variation on the `getToolCategory` method that doesn't require the `Tool` object and takes the category as a string

Reply:
@frett disregard the comments above.